src/lilgit/model/sampler.ts

```diff
@@ -411,4 +470,19 @@ export class Sampler {
 export namespace Sampler {
   export const NO_DIFF = -1;
   export const CHANGE_DIFF = 0;
+  export const PRESENT_DIFF = 1;
+}
+
+export enum SAMPLE_TYPE {
```

Comment:
Where does SAMPLE_TYPE get used and what is its goal?

Reply:
It's used to differentiate types of rendering that should be applied in when calling VersionSampler.sample, to allow different render functions to be used based on type instead of reading excessive optional parameters and casing on undefined.